### PR TITLE
Custom user agent adaditional checks / Intent-Activity refactor ...

### LINKED ISSUE #595: Google Play Store opens after clicking "No"
# Description
When opening a play.google.com url, the app can ask to open the Play Store app. If you click no, the app opens anyway.

# Environment
 - Device: Infinix Hot 10 Lite — X657B
 - OS: Android 10
 - Version: 1.9.17

# To reproduce
Steps to reproduce the behaviour:
1. Have "Launch app" with the "Ask" value in Domain settings
2. Open any play store link
3. Click "No"

# Expected behaviour
It should not open.

# Screenshots

https://github.com/Slion/Fulguris/assets/94485936/6b57bd11-4dbe-484b-8e2a-f202366e3f2a

### PR DESCRIPTION
**Pull request by ZhymabekRoman**
Hi, thanks for the project! I made some changes, to make Fulguris better. Changes are:

 - Additional custom user agent check for new line breaks and tabulator. Fixes:
```
java.lang.IllegalArgumentException: HTTP headers must not contain null, CR, or NL characters. Invalid User-Agent 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36 Edg/122.0.2365.92
```

- Make asking to run an external application by default. This has been a big problem for me. I used Fulguris for a while, but every time I get a weird bug with not opening intents. And once I deleted Fulguris because of that. Now I understand that it was because the apps opening flag was disabled by default in the domain settings.

- Major refactor of the intent logic. Fixes #595. Also can now handle `browser_fallback_url' param. This param value of the URL will be opened if there is no activity available. Some other minor fixes.

### CODE REVIEW

File: app/src/main/java/fulguris/settings/preferences/DomainPreferences.kt [hunk 1]
```diff
@@ -161,7 +161,7 @@ class DomainPreferences constructor(
      * - Ask: Ask the user whether or not to launch the app
      */
     var launchAppOverride by preferences.booleanPreference(R.string.pref_key_launch_app_override, false)
-    var launchAppLocal by preferences.enumPreference(R.string.pref_key_launch_app, NoYesAsk.NO)
+    var launchAppLocal by preferences.enumPreference(R.string.pref_key_launch_app, NoYesAsk.ASK)
```
Comment (Slion) [inline app/src/main/java/fulguris/settings/preferences/DomainPreferences.kt:164]: Default should be set to Yes, see: https://github.com/Slion/Fulguris/issues/598#issuecomment-1985950169

**Comment by Slion** [comment]
Thanks, that's some really good contributions. Sorting out the intent stuff was long overdue. Though ideally we should have one pull request for each topic. I'll try to get this merged this week.

File: app/src/main/java/fulguris/settings/preferences/DomainPreferences.kt (hunk 1 above)
Comment (ZhymabekRoman) [reply app/src/main/java/fulguris/settings/preferences/DomainPreferences.kt:164]: Okay

**Comment by Slion** [comment]
> I can make separate PRs if it is really important. Probably, rebasing can partially solve this problem.

Don't worry for this one. I'll sort it out.

**Comment by Slion** [comment]
How did you test the intent changes? Could you describe the use cases?

**Comment by ZhymabekRoman** [comment]
> How did you test the intent changes? Could you describe the use cases?

- Basic test: https://t.me/durov, https://api.whatsapp.com/send?phone=91
- Testing `browser_fallback_url`: open random Reddit page and press "Use App" button in top of the page. [Example](https://www.reddit.com/r/Damnthatsinteresting/comments/1bj2ff2/no_idea_how_she_was_able_to_keep_a_straight_face/?xpromo_edp=enabled). If domain app settings set to "Yes" it should open Reddit, either it's should open search page in Play Market with Reddit package name. If domain app settings set to "Ask", after pressing "No" in the message box, it's should trigger `browser_fallback_url` page. `browser_fallback_url` is just a Play Market page of the Reddit application.
- https://play.google.com/store/apps/details?id=com.reddit.frontpage - If domain app settings set to "Ask" navigating Play Market page shouldn't trigger intent execution without user promt. After pressing "See in Play store" it's should open Play Market.

**Comment by Slion** [comment]
I merged all changes except the upload of build artefacts as that would result in publishing multiple builds with the same build numbers and I would like to avoid that. If that's something you really need I guess the solution could be to have another build variant. Open an issue about it if you want.

**Pull request closed without merging**